speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ConfirmationMenu.cpp

```diff
@@ -0,0 +1,85 @@
+#include <tgfclient.h>
+
+#include "legacymenu.h"
+#include "ConfigEnums.h"
+#include "mediator.h"
+#include "ConfirmationMenu.h"
+#include "mainmenu.h"
+
+static void* s_menuHandle = nullptr;
+
+/// @brief tells the mediator to not save experiment data and close SpeedDreams
+static void OnAcceptExit(void* /* dummy */)
+{
+    SMediator::GetInstance()->SetSaveRaceToDatabase(false);
+    LmRaceEngine().abortRace();  // Do cleanup to get back correct setup files
+    LegacyMenu::self().quit();
+}
+
+/// @brief tells the mediator to not save experiment data and restart SpeedDreams
+static void OnAcceptRestart(void* /* dummy */)
+{
+    SMediator::GetInstance()->SetSaveRaceToDatabase(false);
+    LmRaceEngine().restartRace();
+}
+
+/// @brief tells the mediator to save experiment data and abort the race
+static void OnAcceptAbort(void* /* dummy */)
+{
+    SMediator::GetInstance()->SetSaveRaceToDatabase(false);
+    LmRaceEngine().abortRace();
+}
+
+/// @brief                  create a confirmation screen
+/// @param p_prevMenu       the previous menu from where it came
+/// @param p_raceEndType enum that decided how you got to the save screen
+void* ConfirmationMenuInit(void* p_prevMenu, RaceEndType p_raceEndType)
+{
+    if (s_menuHandle)
+    {
+        GfuiScreenRelease(s_menuHandle);
+    }
+
+    s_menuHandle = GfuiScreenCreate();
+
+    void* param = GfuiMenuLoad("confirmationmenu.xml");
+    GfuiMenuCreateStaticControls(s_menuHandle, param);
+    switch (p_raceEndType)  ////add different button functionality based on the RaceEndType
```

Comment:
Remove comment or place it above the switch statement
```suggestion
    switch (p_raceEndType)
```